protocol/communities_messenger_test.go

```diff
@@ -3272,49 +3285,269 @@ func (s *MessengerCommunitiesSuite) TestStartCommunityRekeyLoop() {
+		TokenCriteria: []*protobuf.TokenCriteria{{
+			ContractAddresses: map[uint64]string{3: "0x933"},
+			Type:              protobuf.CommunityTokenType_ERC20,
+			Symbol:            "STT",
+			Name:              "Status Test Token",
+			Amount:            "10",
+			Decimals:          18,
+		}},
+	})
+	s.Require().NoError(err)
 
-		c, err = s.admin.GetCommunityByID(c.ID())
-		s.Require().NoError(err)
-		s.Require().True(c.Encrypted())
-
-		// Add Alice and Bob to the community
-		response, err = s.admin.InviteUsersToCommunity(
-			&requests.InviteUsersToCommunity{
-				CommunityID: c.ID(),
-				Users: []types.HexBytes{
-					common.PubkeyToHexBytes(&s.alice.identity.PublicKey),
-					common.PubkeyToHexBytes(&s.bob.identity.PublicKey),
-				},
-			},
-		)
-		s.Require().NoError(err)
-		s.Require().NotNil(response)
-		s.Require().Len(response.Communities(), 1)
+	c, err = s.owner.GetCommunityByID(c.ID())
+	s.Require().NoError(err)
+	s.Require().True(c.Encrypted())
 
-		// Check the Alice and Bob are members of the community
-		c, err = s.admin.GetCommunityByID(c.ID())
-		s.Require().NoError(err)
-		s.Require().True(c.HasMember(&s.alice.identity.PublicKey))
-		s.Require().True(c.HasMember(&s.bob.identity.PublicKey))
+	s.advertiseCommunityTo(c, s.bob, s.owner)
+	s.advertiseCommunityTo(c, s.alice, s.owner)
+
+	s.owner.communitiesManager.PermissionChecker = &testPermissionChecker{}
+
+	s.joinCommunity(c, s.bob, s.owner)
+	s.joinCommunity(c, s.alice, s.owner)
+
+	// Check the Alice and Bob are members of the community
+	c, err = s.owner.GetCommunityByID(c.ID())
+	s.Require().NoError(err)
+	s.Require().True(c.HasMember(&s.alice.identity.PublicKey))
+	s.Require().True(c.HasMember(&s.bob.identity.PublicKey))
+
+	// Check the keys in the database
+	keys, err := s.owner.sender.GetKeysForGroup(c.ID())
+	s.Require().NoError(err)
+	keyCount := len(keys)
 
-		// Check the keys in the database
-		keys, err := s.admin.sender.GetKeyIDsForGroup(c.ID())
+	// Check that rekeying is occurring by counting the number of keyIDs in the encryptor's DB
+	// This test could be flaky, as the rekey function may not be finished before RekeyInterval * 2 has passed
+	for i := 0; i < 5; i++ {
+		time.Sleep(s.owner.communitiesManager.RekeyInterval * 2)
+		keys, err = s.owner.sender.GetKeysForGroup(c.ID())
 		s.Require().NoError(err)
-		keyCount := len(keys)
+		s.Require().Greater(len(keys), keyCount)
+		keyCount = len(keys)
+	}
+}
 
-		// Check that rekeying is occurring by counting the number of keyIDs in the encryptor's DB
-		// This test could be flaky, as the rekey function may not be finished before RekeyInterval * 2 has passed
-		for i := 0; i < 5; i++ {
-			time.Sleep(s.admin.communitiesManager.RekeyInterval * 2)
-			keys, err = s.admin.sender.GetKeyIDsForGroup(c.ID())
-			s.Require().NoError(err)
-			s.Require().Greater(len(keys), keyCount)
-			keyCount = len(keys)
-		}
-	*/
+func (s *MessengerCommunitiesSuite) TestCommunityRekeyAfterBan() {
+	admin := s.newMessenger()
```

Comment:
I'm curious, why can't you use the basic `s.owner`?

Reply:
some tests wants an aggressive rekey interval in the loop, while this one wants it disabled, and it's done on creation (before start), so I though the best would be to create a new one